_posts/2019-12-08-what-is-a-shell.md

```diff
@@ -0,0 +1,166 @@
+If you have ever interacted with MacOS/Linux *terminal* or CMD and PowerShell on Windows, you're using what most users will call "the shell".
+Technically, the program that opens up the window you're typing into is most likley a terminal emulator,
+but the program running inside accepting your input is the shell.
+
+An operating system shell allows you interact with programs installed on the operating system which in turn operate on the file system, 
+system processes, devices that make up the system you're on, and much more.
+In some cases, you are able to communicate with completely different machines as over SSH, aka Secure Shell.
+Software developers and IT professional use shells for many tasks such as batch processing files, searching the file system, running custom scripts, downloading the uploading files to the internet, automating their daily workflows, and even playing games!
+
+Some examples of well known shells are Bash, Zsh, Fish, Powershell, Korn Shell, and csh.
+There are many others (I'm sorry if I didn't mention your favorite one!), but listing them all would belabor the point.
+
+### Example Usage
+
+In the [Bash shell](https://en.wikipedia.org/wiki/Bash_%28Unix_shell%29) (bourne again shell), 
+you can list directory contents using the command `ls`,
+where it displays information about the current directory by default.
+[Powershell](https://docs.microsoft.com/en-us/powershell/) on Windows provides users a similar functionality using the same `ls` command.
+
+In order to alter the way information is displayed,
+you may have to provide additional input to the command,
+which is called a **flag**.
+For example, `ls -l` provides a "long list" format which also shows file permissions, file creation and file access information.
+That particular flag will work in Bash and most other Bash inspired shells, but it doesn't work in Powershell.
+
+Powershell provides a similar features in the form of a cmdlet, pronounced _command-let_.
+You can try `ls` in Powershell and it will work, but also try `man ls` to get the manual page.
+You'll find that the cmdlet is actually named `Get-ChildItem`.
+If you use `alias ls` in Powershell,
+you will see the command is an alias for the Get-ChildItem.
+
+It's somewhat interesting that `ls` in Bash is similar to `ls -Name` in Powershell
+and `ls -l` in Bash is similar to the default `ls` in Powershell.
+
+There are plenty of other executable programs that operating systems have to offer.
+For Linux/MacOS,
+you most likely have the path `/usr/bin` in your `PATH` variable,
+which is what directories your shell will look in to find executable programs.
+Try `ls -l /usr/bin` to see a list of executables which you can run from the command line.
+If you're curious about any of them,
+all you need is the manual page by typing the executable name preceded by `man`.
+For example `man zip`.
+Similar functionality can be achieved in Powershell by using `Get-Command`.
```

Comment:
Not sure if you want to call out some other common default paths. Here's what I have on OSX bash:

```
$ env -i bash --noprofile --norc -c '/bin/echo $PATH'
/usr/gnu/bin:/usr/local/bin:/bin:/usr/bin:.
```